packages/big-design/src/utils/messagingHelpers.tsx

```diff
@@ -0,0 +1,63 @@
+import { CheckCircleIcon, ErrorIcon, InfoIcon, WarningIcon } from '@bigcommerce/big-design-icons';
+import { ThemeInterface } from '@bigcommerce/big-design-theme';
+import React from 'react';
+import { css } from 'styled-components';
+
+import { LinkProps } from '../components/Link';
+
+export interface SharedMessagingProps extends React.HTMLAttributes<HTMLDivElement> {
+  header?: string;
+  messages: MessageItem[];
+  type?: MessagingType;
+  onClose?(): void;
+}
+
+export type MessagingType = 'success' | 'error' | 'warning' | 'info';
+
+export type MessagingVariant = 'alert' | 'message' | 'inline';
+
+export interface MessageItem {
+  text: string;
+  link?: MessageLinkItem;
+}
+
+export type MessageLinkItem = Pick<LinkProps, 'external' | 'href' | 'target'> & {
+  text: string;
+};
+
+export const getMessagingIcon = (type: MessagingType, condensed?: boolean) => {
```

Comment:
Wherever this function ise used, make sure to memo the call. Icons are expensive.